controllers/openstackdataplane_controller.go

```diff
@@ -62,5 +218,97 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 func (r *OpenStackDataPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&dataplanev1beta1.OpenStackDataPlane{}).
+		Owns(&dataplanev1beta1.OpenStackDataPlaneNode{}).
+		Owns(&dataplanev1beta1.OpenStackDataPlaneRole{}).
 		Complete(r)
 }
+
+// CreateDataPlaneResources -
+func CreateDataPlaneResources(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper) (ctrl.Result, error) {
+	// roles and nodes may depend on each other, so we run both before checking for errors
+	roleErr := CreateDataPlaneRole(ctx, instance, helper)
+	nodeErr := CreateDataPlaneNode(ctx, instance, helper)
+	if roleErr != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			dataplanev1beta1.DataPlaneRoleErrorMessage,
+			roleErr.Error()))
+		return ctrl.Result{}, roleErr
+	}
+	if nodeErr != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			dataplanev1beta1.DataPlaneNodeErrorMessage,
+			nodeErr.Error()))
+		return ctrl.Result{}, nodeErr
+	}
+
+	return ctrl.Result{}, nil
+
+}
+
+// CreateDataPlaneNode -
+func CreateDataPlaneNode(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper) error {
+	client := helper.GetClient()
+	logger := helper.GetLogger()
+	node := &dataplanev1beta1.OpenStackDataPlaneNode{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: instance.Namespace,
+		},
+	}
+
+	for nodeName, nodeSpec := range instance.Spec.Nodes {
+		node.Name = nodeName
+		nodeSpec.DeepCopyInto(&node.Spec)
+		err := client.Get(ctx, types.NamespacedName{Name: node.Name, Namespace: instance.Namespace}, node)
+		if err != nil && !k8s_errors.IsNotFound(err) {
+			logger.Info("Failed to get Node", "Node.Namespace", instance.Namespace, "Node.Name", node.Name)
+			return err
+		}
+		if k8s_errors.IsNotFound(err) {
+			err := client.Create(ctx, node)
+			if err != nil {
+				logger.Info("Failed to create Node", "Node.Namespace", instance.Namespace, "Node.Name", node.Name)
+				return err
```

Comment:
I am getting an error here:

'''
2023-03-17T13:02:34-04:00       INFO    controllers.OpenStackDataPlane  Failed to create Node   {"Node.Namespace": "openstack", "Node.Name": "edpm-compute-1"}               2023-03-17T13:02:38-04:00       ERROR   Reconciler error        {"controller": "openstackdataplane", "controllerGroup": "dataplane.openstack.org", "controllerKind": "OpenStackDataPlane", "OpenStackDataPlane": {"name":"openstack-edpm","namespace":"openstack"}, "namespace": "openstack", "name": "openstack-edpm", "reconcileID": "dbe7510e-f1fb-4b5b
-ae71-f8c97aa2b3d8", "error": "resourceVersion should not be set on objects to be created"}                                                                                  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /home/jslagle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:329                                                            
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem                                                                                             /home/jslagle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2                                                                                           
        /home/jslagle/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235                                                            2023-03-17T13:02:38-04:00       INFO    controllers.OpenStackDataPlane  CreateDataPlaneResources
'''